### PR TITLE
Reduce number of fixtures created for pagination tests

### PR DESCRIPTION
**Pull request by cwrw**
Set a different Kaminari page default in tests. Where previously it equated to 20, set it to 5 instead and amend tests to create less fixtures. 

Also move pagination tests to a shared example.

### CODE REVIEW

File: spec/requests/api/v1/journeys_controller_index_spec.rb [hunk 1]
```diff
@@ -22,11 +22,12 @@
...
     let(:intermediate_journeys_count) { 1 }
     let(:page) { 1 }
-    let(:per_page) { 20 }
+    let(:per_page) { 5 }
     let(:url) { "/api/v1/moves/#{move.id}/journeys" }
+    let(:params) { {} }
 
     before do
-      get url, headers: headers, as: :json
+      get url, params: params, headers: headers
```
Comment (cwrw) [inline spec/requests/api/v1/journeys_controller_index_spec.rb:30]: when attempting to pass in `params`, the `as: :json` was changing the `GET` request to a `POST`

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -89,3 +89,7 @@
     with.library :rails
   end
 end
+
+Kaminari.configure do |config|
```
Comment (smoothcontract) [inline spec/rails_helper.rb:93]: Might be nice to add a comment explaining why this overrides the default only affects tests.

File: spec/requests/api/v1/moves_controller_index_spec.rb [hunk 3]
```diff
@@ -107,29 +107,9 @@
           }
         end
 
-        it 'paginates 20 results per page as default' do
-          get_moves
-
-          expect(response_json['data'].size).to eq 20
-        end
-
-        it 'returns 1 result on the second page'  do
```
Comment (smoothcontract) [inline spec/requests/api/v1/moves_controller_index_spec.rb:116]: Looks like this test (of the links) is now removed. Is there a way that could be added to the shared context, perhaps by reading the link from the returned metadata?

Comment (cwrw) [reply spec/requests/api/v1/moves_controller_index_spec.rb:116]: talked in person but for posterity: this test is defined in the shared example but using `params` instead

**Pull request merged**